### PR TITLE
Add backend reference

### LINKED ISSUE #364: Mention --log-level flag for the backend and agent in Sensu 2.x
Recently, a `--log-level` flag was introduced in sensu-go. The documentation should be updated accordingly.

### LINKED ISSUE #528: Backend reference needed
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)

## Expected Behavior

Documentation should be comprehensive.

## Current Behavior

Backend reference documentation is not available for 2.0

## Possible Solution

Document backend behaviors for 2.0, similar to the server reference doc for 1.x

## Context

In working to update some of our "getting started" email templates we've found that there's no analog for the 1.x server reference document in 2.0 documentation.

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

Adds a starting point for a backend reference doc, including:

- Feature overview
- Commands
- Configuration flags
  - Current flag descriptions are copied from the `sensu-backend` help content.
  - Examples comes from clustering guide and example config file in sensu-go
  - Defaults come from `sensu-backend` help content and https://github.com/sensu/sensu-go/blob/master/backend/cmd/start.go

## Motivation and Context

Closes #364. Closes #528.

### CODE REVIEW

File: content/sensu-core/2.0/reference/backend.md [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+example     | {{< highlight shell >}}# Command line example
+sensu-backend start --debug
+
+# /etc/sensu/backend.yml example
+debug: true{{< /highlight >}}
+
+
+| deregistration-handler |      |
+-------------------------|------
+description              | Default deregistration handler
```
Comment (echlebek) [inline content/sensu-core/2.0/reference/backend.md]: Could the description here explain what a deregistration handler is for?

File: content/sensu-core/2.0/reference/backend.md [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+example           | {{< highlight shell >}}# Command line example
+sensu-backend start --trusted-ca-file /path/to/trusted-certificate-authorities.pem
+
+# /etc/sensu/backend.yml example
+trusted-ca-file: "/path/to/trusted-certificate-authorities.pem"{{< /highlight >}}
+
+
+| insecure-skip-tls-verify |      |
+---------------------------|------
+description                | skip ssl verification
```
Comment (echlebek) [inline content/sensu-core/2.0/reference/backend.md]: The name says it all, but it might be nice to mention here that this shouldn't be used in production.

**Review by echlebek — APPROVED** [review]
This looks great. To the best of my knowledge, what is written here is true and correct! I have a couple small comments.

:shipit:

File: content/sensu-core/2.0/reference/backend.md [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+  - [General configuration](#general-configuration-flags)
+  - [Agent communication configuration](#agent-communication-configuration-flags)
+  - [Security configuration](#security-configuration-flags)
+  - [Dashboard configuration](#dashboard-configuration-flags)
+  - [Datastore and cluster configuration](#datastore-and-cluster-configuration-flags)
+
+## How does the backend work?
+
+The Sensu backend is a service that manages check requests and event data.
+You can run the backend as a standalone service, but running a cluster of backends makes Sensu more highly available, reliable, and durable.
```
Comment (rgeniesse) [inline content/sensu-core/2.0/reference/backend.md]: Maybe link to the clustering guide here or do we want to keep that out of references?

File: content/sensu-core/2.0/reference/backend.md [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+
+# /etc/sensu/backend.yml example
+log-level: "debug"{{< /highlight >}}
+
+
+| state-dir  |      |
+-------------|------
+description  | Path to Sensu state storage 
+type         | String
+default      | <ul><li>Windows: `C:\\ProgramData\sensu\data`</li><li>FreeBSD: `/usr/local/etc/sensu/backend.yml`</li><li>Linux: `/var/lib/sensu`</li></ul>
```
Comment (rgeniesse) [inline content/sensu-core/2.0/reference/backend.md]: For Windows and Linux, these refer to directories. For FreeBSD, it is a file. Should it be `/usr/local/etc/sensu` as this section is for `state-dir`?

File: content/sensu-core/2.0/reference/backend.md (hunk 3 above)
Comment (rgeniesse) [inline content/sensu-core/2.0/reference/backend.md]: Probably a nitpick, but the mention of clustering feels out of place. Possibly a link here (there is one a bit further down) or not mentioning clustering until later?

**Comment by apaskulin** [comment]
Thank you, @echlebek and @rgeniesse! 💚  I've updated the PR to incorporate your comments.

**Pull request merged**